### PR TITLE
Add a WebSocket++ fingerprint

### PR DESCRIPTION
**Pull request by dabdine**
## Description
Adds a fingerprint for WebSocket++. There are over 72K of these on the internet, [according to Censys](https://search.censys.io/search?resource=hosts&sort=RELEVANCE&per_page=25&virtual_hosts=EXCLUDE&q=same_service%28not+services.software.product%3A*+and+services.service_name%3A+HTTP+and+services.http.response.headers.server%3AWebSocket%2B%2B*%29).

The server git repository is on GH:
https://github.com/zaphoyd/websocketpp

## Motivation and Context
Adds a fingerprint

## How Has This Been Tested?
`bundle exec rake tests`

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: xml/http_servers.xml [hunk 1]
```diff
@@ -1506,6 +1506,20 @@
...
+  <fingerprint pattern="^WebSocket\+\+/([\d.-]+(?:-dev)?)$">
+    <description>WebSocket++ web server - https://github.com/zaphoyd/websocketpp</description>
+    <example service.version="0.8.1">WebSocket++/0.8.1</example>
+    <example service.version="0.8.2">WebSocket++/0.8.2</example>
+    <example service.version="0.4.0">WebSocket++/0.4.0</example>
+    <example service.version="0.8.0-dev">WebSocket++/0.8.0-dev</example>
+    <example service.version="2011-09-25">WebSocket++/2011-09-25</example>
+    <param pos="0" name="service.vendor" value="Zaphoyd Studios"/>
+    <param pos="0" name="service.product" value="WebSocket++"/>
+    <param pos="0" name="service.family" value="WebSocket++"/>
```
Comment (mkienow-r7) [inline xml/http_servers.xml:1518]: Is WebSocket++ a `service.family`?

Comment (dabdine) [reply xml/http_servers.xml:1518]: Good point, will remove

File: xml/http_servers.xml [hunk 2]
```diff
@@ -1506,6 +1506,19 @@
     <param pos="0" name="service.cpe23" value="cpe:/a:nginx:nginx:{service.version}"/>
   </fingerprint>
 
+  <fingerprint pattern="^WebSocket\+\+/([\d.-]+(?:-dev)?)$">
+    <description>WebSocket++ web server - https://github.com/zaphoyd/websocketpp</description>
+    <example service.version="0.8.1">WebSocket++/0.8.1</example>
+    <example service.version="0.8.2">WebSocket++/0.8.2</example>
+    <example service.version="0.4.0">WebSocket++/0.4.0</example>
+    <example service.version="0.8.0-dev">WebSocket++/0.8.0-dev</example>
+    <example service.version="2011-09-25">WebSocket++/2011-09-25</example>
```
Comment (mkienow-r7) [inline xml/http_servers.xml:1515]: The date value doesn't appear to be valid version of this software. See zaphoyd/websocketpp
[changelog.md](https://github.com/zaphoyd/websocketpp/blob/master/changelog.md)
> 0.3.0-alpha1 - 2013-06-09
> - Initial Release

Comment (dabdine) [reply xml/http_servers.xml:1515]: It's a live server header via censys

Comment (dabdine) [reply xml/http_servers.xml:1515]: https://search.censys.io/search?resource=hosts&sort=RELEVANCE&per_page=25&virtual_hosts=EXCLUDE&q=same_service%28not+services.software.product%3A*+and+services.service_name%3A+HTTP+and+services.http.response.headers.server%3AWebSocket%2B%2B%2F2011-09-25%29

Comment (mkienow-r7) [reply xml/http_servers.xml:1515]: Do we have any additional evidence it is actually WebSocket++?

Comment (dabdine) [reply xml/http_servers.xml:1515]: Although all of them are in China, which often has questionable server values (such as "iis8.0"). It's possible they're honeypots or simply faked for whatever reason.

Comment (mkienow-r7) [reply xml/http_servers.xml:1515]: I took a deep dive into the commit history of zaphoyd/websocketpp and found `WebSocket++/2011-09-25` hardcoded! 
https://github.com/zaphoyd/websocketpp/blob/0568bbe1126478fadeeb52f7fafe6ed49608b23f/src/websocket_server_session.cpp#L285

**Comment by mkienow-r7** [comment]
Would you please rebase with master to resolve the merge conflict?

**Review by mkienow-r7 — APPROVED** [review]
I overlooked that the `service.vendor` and `service.product` values do not map back to the CPE values. When we run `update_cpes.py` it will end up removing the `service.cpe23` param. I will merge this as-is and open a small PR to change those values.

Thank you for the contribution @dabdine!

**Comment by dabdine** [comment]
@mkienow-r7 thanks for the reviews and shepherding the merge!

**Pull request merged**